### ISSUE TITLE
dropout rates showing as integers in view of runs

### ISSUE THREAD

**Issue by medewitt** [rstudio/tfruns#21]
I have trained a model over various parameters (activation functions, drop out rates, etc). On the View of the output the drop out rates appear as zero in the shiny popup when invoking the cloudml::job_collect().

![image](https://user-images.githubusercontent.com/25038837/35570825-4f53f552-059e-11e8-9849-00e229e657f5.png)

The dropout rates show correctly in the job_trials() output (as numerics with values)...I think it is whatever creates the flags.json file during job collection is the culprit as it shows the drop_outs as zero:

flags.json:
![image](https://user-images.githubusercontent.com/25038837/35571245-8cc84eaa-059f-11e8-86b5-900b648c1733.png)

Maybe type conversion on ls_runs.R lines 280-300? Wish I could be of more help...

**Comment by jjallaire** [comment]
I just tried to repro this but couldn't (either locally with `tfruns::training_run()` or with `cloudml::cloudml_train()`). 

I recall a bug similar to this a few weeks back. What happens if you update to the latest version of the cloudml package from GitHub?

If that doesn't fix it then if you could provide me with the exact code you are using (or a minimally reproducible example based on it) I can investigate further.

**Comment by medewitt** [comment]
Updated to latest version of cloudml. I also found I made an error naming my flags in the R code (flag_integer instead of flag_numeric for drop out). However, learning rate still is appearing as a zero in the output.

![image](https://user-images.githubusercontent.com/25038837/35648836-0c66697a-06a5-11e8-98a7-2f24896e0257.png)

x_train <- matrix(rnorm(1000*50, 30, 5), nrow = 1000, ncol = 50)
y_train <- matrix(rbinom(1000, 1,.3), nrow = 1000, ncol = 1)


_R code:_
FLAGS <- flags(
  flag_string("activation_1", "relu", "One of relu, selu, tanh, sigmoid"),
  flag_string("activation_2", "relu", "One of relu, selu, tanh, sigmoid"),
  flag_string("activation_3", "relu", "One of relu, selu, tanh, sigmoid"),
  flag_numeric("learning_rate", 0.001, "Optimizer Learning Rate"),
  flag_integer("hidden_size_1", 15, "The hidden layer size"),
  flag_integer("hidden_size_2", 15, "The hidden layer size"),
  flag_integer("hidden_size_3", 15, "The hidden layer size"),
  flag_numeric("dropout_1", .2, "The dropout for layer 1"),
  flag_numeric("dropout_2", .2, "The dropout for layer 2"),
  flag_numeric("dropout_3", .2, "The dropout for layer 3")
)

#Build model ------
model <- keras_model_sequential()
model %>%
  layer_dense(FLAGS$hidden_size_1, activation = FLAGS$activation_1, input_shape = ncol(x_train)) %>%
  layer_dropout(FLAGS$dropout_1) %>% 
  layer_dense(FLAGS$hidden_size_2, activation = FLAGS$activation_2) %>%
  layer_dropout(FLAGS$dropout_2) %>% 
  layer_dense(FLAGS$hidden_size_3, activation = FLAGS$activation_3) %>%
  layer_dropout(FLAGS$dropout_3) %>% 
  layer_dense(units = 1, activation = "sigmoid")

summary(model)

model %>% compile(
  optimizer = optimizer_adam(lr = FLAGS$learning_rate),
  loss = loss_binary_crossentropy,
  metrics = c('accuracy')
) 

checkpoint <- callback_model_checkpoint(
  filepath = "model_adam_sequential.hdf5", 
  save_best_only = TRUE, 
  period = 1,
  verbose = 1
)

early_stopping <- callback_early_stopping(patience = 5)

model %>% fit(
  x_train, y_train, 
  epochs = 100, 
  batch_size = 32, 
  callbacks = list(checkpoint, early_stopping),
  validation_split = 0.4
)

_Config_

trainingInput:
  scaleTier: CUSTOM
  masterType: standard_gpu
  hyperparameters:
    goal: MAXIMIZE
    hyperparameterMetricTag: val_acc
    maxTrials: 100
    maxParallelTrials: 5
    params:
      - parameterName: activation_1
        type: CATEGORICAL
        categoricalValues: [relu, selu, tanh, sigmoid]
      - parameterName: activation_2
        type: CATEGORICAL
        categoricalValues: [relu, selu, tanh, sigmoid]
      - parameterName: activation_3
        type: CATEGORICAL
        categoricalValues: [relu, selu, tanh, sigmoid]
      - parameterName: learning_rate
        type: DOUBLE
        minValue: 0.000001
        maxValue: 0.1
        scaleType: UNIT_LOG_SCALE
      - parameterName: hidden_size_1
        type: INTEGER
        minValue: 5
        maxValue: 50
        scaleType: UNIT_LINEAR_SCALE
      - parameterName: hidden_size_2
        type: INTEGER
        minValue: 5
        maxValue: 50
        scaleType: UNIT_LINEAR_SCALE
      - parameterName: hidden_size_3
        type: INTEGER
        minValue: 5
        maxValue: 50
        scaleType: UNIT_LINEAR_SCALE
      - parameterName: dropout_1
        type: DOUBLE
        minValue: .1
        maxValue: .50
        scaleType: UNIT_LINEAR_SCALE
      - parameterName: dropout_2
        type: DOUBLE
        minValue: .1
        maxValue: .50
        scaleType: UNIT_LINEAR_SCALE
      - parameterName: dropout_3
        type: DOUBLE
        minValue: .1
        maxValue: .50
        scaleType: UNIT_LINEAR_SCALE

**Comment by jjallaire** [comment]
Using your code exactly I still get the correct learning rate reported when running both a local `training_run()` and when doing a single `cloudml_train()` (in both cases I passed  0.000001 explicitly as the `learning_rate`).

What happens when you try those simpler scenarios? (I'm thinking that perhaps there is some truncation going on when CloudML passes us hyperparameters?)

I'm currently trying with your full tuning yaml on cloudml to see what I get there.

**Comment by medewitt** [comment]
I ran it locally with` training_run()` and  on cloudML with the simple datasets and it works correctly with the toy data. I ran it again with my real data and the learning rate again appears as zero...must be some feature of my data?

**Comment by jjallaire** [comment]
I can't imagine what about your data would cause this though. Is there any way to share enough of it with me that I can reproduce locally?

**Comment by medewitt** [comment]
I think we can close this one. I reran everything with tfruns and cloudml::cloudml_train with my actual data after I updated the google cloud SDK and everything is appearing as it should. Maybe that was the root cause (something amiss in google cloud not returning the learning rate).
